### PR TITLE
Active window watching mechanism implemented

### PR DESCRIPTION
**Pull request by helen-dikareva**
It's necessary to fix the T235186 TestCafe bug

PR is clone of https://github.com/superroma/testcafe-hammerhead/pull/49

/cc @miherlosev @inikulin

@VasilyStrelyaev please check tests names

### CODE REVIEW

File: test/client/fixtures/sandboxes/event/focus-blur-change-test.js [hunk 1]
```diff
@@ -753,6 +755,29 @@ asyncTest('blur() must not raise event if element is already blured', function (
     );
 });
 
+asyncTest('active window doesn\'t change after focus ShadowUI element in iFrame', function () {
```
Comment (VasilyStrelyaev) [inline test/client/fixtures/sandboxes/event/focus-blur-change-test.js]: after focusing

File: test/client/fixtures/sandboxes/event/active-window-tracker-test.js [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+                ok(ActiveWindowTracker.getState());
+                notOk(iFrameWindow.ActiveWindowTracker.getState());
+
+                $iFrame.remove();
+                start();
+            });
+    });
+});
+
+asyncTest('check changing active window (between two iFrames)', function () {
```
Comment (VasilyStrelyaev) [inline test/client/fixtures/sandboxes/event/active-window-tracker-test.js]: changing ... between -> switching ... between

**Comment by VasilyStrelyaev** [comment]
\r-

**Comment by helen-dikareva** [comment]
FPR
tests passed https://travis-ci.org/testcafe-build-bot/testcafe-hammerhead/builds/76637931

**Comment by inikulin** [comment]
\cc @AlexanderMoskovkin Yet another strange case with our bot

**Comment by inikulin** [comment]
lgtm

**Comment by miherlosev** [comment]
lgtm.
\cc @VasilyStrelyaev - check and merge

**Pull request merged**